graylog2-web-interface/src/stores/sidecars/ConfigurationVariableStore.js

```diff
@@ -0,0 +1,114 @@
+import Reflux from 'reflux';
+import URI from 'urijs';
+
+import URLUtils from 'util/URLUtils';
+import UserNotification from 'util/UserNotification';
+import fetch from 'logic/rest/FetchProvider';
+import CombinedProvider from 'injection/CombinedProvider';
+
+const { ConfigurationVariableActions } = CombinedProvider.get('ConfigurationVariable');
+
+const ConfigurationVariableStore = Reflux.createStore({
+  listenables: [ConfigurationVariableActions],
+  sourceUrl: '/sidecar/configuration_variables',
+
+  _fetchConfigurationVariables() {
+    const baseUrl = `${this.sourceUrl}`;
+    const uri = URI(baseUrl).toString();
+
+    return fetch('GET', URLUtils.qualifyUrl(uri));
+  },
+
+  all() {
+    const promise = this._fetchConfigurationVariables();
+    promise
+      .then(
+        () => {},
+        (error) => {
+          UserNotification.error(`Fetching configuration variables failed with status: ${error}`,
+            'Could not retrieve configuration variables');
+        });
+
+    ConfigurationVariableActions.all.promise(promise);
+  },
+
+  save(configurationVariable) {
+    const request = {
+      id: configurationVariable.id,
+      name: configurationVariable.name,
+      description: configurationVariable.description,
+      content: configurationVariable.content,
+    };
+
+    let url = URLUtils.qualifyUrl(`${this.sourceUrl}`);
+    let method;
+    if (configurationVariable.id === '') {
+      method = 'POST';
+    } else {
+      url += `/${configurationVariable.id}`;
+      method = 'PUT';
+    }
+
+    const promise = fetch(method, url, request);
+    promise
+      .then(() => {
+        const action = configurationVariable.id === '' ? 'created' : 'updated';
```

Comment:
I think setting this variable in the if above would make it clearer, as all values depending on the type of action would be set in the same place.